chia/_tests/core/full_node/test_address_manager.py

```diff
@@ -526,13 +534,35 @@ async def test_addrman_eviction_works(self):
         await addrman.resolve_tried_collisions()
         assert await addrman.select_tried_collision() is None
 
+    async def check_retrieved_peers(self, wanted_peers: list[ExtendedPeerInfo], addrman: AddressManager) -> bool:
+        retrieved_peers = []
+        for _ in range(50):
+            peer = await addrman.select_peer()
+            if peer not in retrieved_peers:
+                retrieved_peers.append(peer)
+            if len(retrieved_peers) == 3:
+                break
+        if len(retrieved_peers) != 3:
+            return False
```

Comment:
does this always indicate a test failure? if so, I think it's reasonable to add a `pragma: no cover`

Reply:
done